### ISSUE TITLE
UnhandledPromiseRejectionWarning: Error: Invalid name: "redux-saga/effects"

### ISSUE THREAD

**Issue by aulisius** [mastilver/dynamic-cdn-webpack-plugin#32]
Getting this error, when trying to include this plugin into my webpack config.

**Comment by aulisius** [comment]
Fixed this currently by using the `only` option passed to the plugin, but should the plugin break when it can't resolve a package?

**Comment by mastilver** [comment]
> should the plugin break when it can't resolve a package?

No, it shouldn't... it should just ignore it... :/

I will need to have a look at that later

**Comment by aulisius** [comment]
I believe adding a `.catch` block [here](https://github.com/mastilver/dynamic-cdn-webpack-plugin/blob/master/src/index.js#L64) should at least prevent the plugin from dying.

**Comment by mastilver** [comment]
Can you give me the full stack trace?

Did you install `reduc-saga`?

**Comment by aulisius** [comment]
This is when I run without using the `only` option.
```
(node:13924) UnhandledPromiseRejectionWarning: Unhandled promise rejection (rejection id: 2): Error: Invalid name: "redux-saga/effects"
(node:13924) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, prom
ise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
(node:13924) UnhandledPromiseRejectionWarning: Unhandled promise rejection (rejection id: 4): Error: Invali
d name: "redux-saga/effects"
(node:13924) UnhandledPromiseRejectionWarning: Unhandled promise rejection (rejection id: 6): Error: Invali
d name: "redux-saga/effects"
(node:13924) UnhandledPromiseRejectionWarning: Unhandled promise rejection (rejection id: 8): Error: Invali
d name: "redux-saga/effects"
(node:13924) UnhandledPromiseRejectionWarning: Unhandled promise rejection (rejection id: 10): Error: Inval
id name: "redux-saga/effects"
(node:13924) UnhandledPromiseRejectionWarning: Unhandled promise rejection (rejection id: 12): Error: Inval
id name: "redux-saga/effects"
(node:13924) UnhandledPromiseRejectionWarning: Unhandled promise rejection (rejection id: 14): Error: Inval
id name: "redux-saga/effects"
(node:13924) UnhandledPromiseRejectionWarning: Unhandled promise rejection (rejection id: 16): Error: Inval
id name: "redux-saga/effects"
(node:13924) UnhandledPromiseRejectionWarning: Unhandled promise rejection (rejection id: 18): Error: Inval
id name: "redux-saga/effects"
(node:13924) UnhandledPromiseRejectionWarning: Unhandled promise rejection (rejection id: 20): Error: Inval
id name: "redux-saga/effects"
(node:13924) UnhandledPromiseRejectionWarning: Unhandled promise rejection (rejection id: 22): Error: Inval
id name: "redux-saga/effects"
(node:13924) UnhandledPromiseRejectionWarning: Unhandled promise rejection (rejection id: 24): Error: Inval
id name: "redux-saga/effects"
(node:13924) UnhandledPromiseRejectionWarning: Unhandled promise rejection (rejection id: 26): Error: Inval
id name: "redux-saga/effects"
(node:13924) UnhandledPromiseRejectionWarning: Unhandled promise rejection (rejection id: 28): Error: Inval
id name: "redux-saga/effects"
(node:13924) UnhandledPromiseRejectionWarning: Unhandled promise rejection (rejection id: 31): Error: Inval
id name: "redux-saga/effects"
(node:13924) UnhandledPromiseRejectionWarning: Unhandled promise rejection (rejection id: 34): Error: Inval
id name: "redux-saga/effects"
Node#moveTo was deprecated. Use Container#append.
(node:13924) UnhandledPromiseRejectionWarning: Unhandled promise rejection (rejection id: 36): Error: Inval
id name: "redux-saga/effects"
(node:13924) UnhandledPromiseRejectionWarning: Unhandled promise rejection (rejection id: 39): Error: Inval
id name: "redux-saga/effects"
(node:13924) UnhandledPromiseRejectionWarning: Unhandled promise rejection (rejection id: 41): Error: Inval
id name: "redux-saga/effects"
(node:13924) UnhandledPromiseRejectionWarning: Unhandled promise rejection (rejection id: 43): Error: Inval
id name: "redux-saga/effects"
```

**Comment by mastilver** [comment]
It reminds me of this issue: https://github.com/benmosher/eslint-plugin-import/issues/793

Is it working at all with `redux-saga@0.14.4`?

**Comment by aulisius** [comment]
I didn't get that error with `redux-saga@0.14.4`.

**Comment by mastilver** [comment]
That's it then! :)

Easier to write a test with that information. There is a good chance, it's due to an underlying dependency (I have no idea which one yet). Maybe https://github.com/sindresorhus/resolve-pkg or https://github.com/sindresorhus/read-pkg-up

**Comment by mastilver** [comment]
For reference: the issue is caused by this change in `redux-saga`: https://github.com/redux-saga/redux-saga/pull/895

**Comment by mastilver** [comment]
I found where the error comes from: https://github.com/npm/normalize-package-data/blob/master/lib/fixer.js#L335

**Comment by mastilver** [comment]
fixed on `v3.4.1`

**Comment by aulisius** [comment]
Awesome. Thanks a lot 💯
